core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -87,7 +87,7 @@ private[spark] class TaskSetManager(
   // Set the coresponding index of Boolean var when the task killed by other attempt tasks,
   // this happened while we set the `spark.speculation` to true. The task killed by others
   // should not resubmit while executor lost.
-  private val killedByOtherAttempt: Array[Boolean] = new Array[Boolean](numTasks)
+  private val killedByOtherAttempt = new HashSet[Long]
```

Comment:
super nit: I prefer a `Array[Long]`, so you know the index corresponding to the taskId, that can provide more information while debug.

Reply:
Hi @jiangxb1987, thanks for the comment, but I'm sure if I understand your suggestion correctly. Do you mean: private val killedByOtherAttempt = new Array[Long] ?

Reply:
Also, the comment "Set the corresponding index of Boolean var when the task killed ..." is not correct anymore. I'm sorry I forgot to update it.

Reply:
Yea, please also update the comment.

Reply:
I think we should use ArrayBuffer[Long] instead of Array[Long] because the number of elements can grow when there are more killed attempts.
Also, I think there is a downside of using Array-like data structure for this variable. Lookup operation for array-like data structure takes linear time and that operation is used many times when we check if a task need to be resubmitted (inside executorLost method of TSM). This will not matter much if the size of the array is small, but still I think this is something we might want to consider.

Reply:
@tgravescs Actually I thought HashSet meant we pay the cost of maintaining this data only when speculative exec is enabled and there are kill's - not in other cases.
If we need the partition for the tasks which were killed, we could make it a HashMap[Long, Int] for [taskId, partition] to efficiently get to the info at minimal memory footprint. What do you think ?

Reply:
I think the hashset is fine, we shouldn't be keeping a huge array if we don't need it. We use to much memory as it is.  @jiangxb1987  how big of a deal is this to you?  I prefer the hashMap approach @mridulm mentions if that is the useful information to you.

Reply:
@mridulm 's approach also sounds good to me.

Reply:
@jiangxb1987  please clarify is it fine as is or are you wanting to use a hashMap and track the index?  Can you give an example when this is used for debugging?  For instance are you getting a heap dump and looking at the datastructures that might make sense,  otherwise its not accessible without you adding in further log statements anyway and its just extra memory usage.

Reply:
I'm not proposing to expand the logInfo in L735 in this PR, I'm just concern about whether it's convenient enough for me to add extra logs to debug a potential issue. Since there is another way to achieve the same effect, I'm okay with using hashSet here.

Reply:
Great thanks, seems like we can go with code as is then.